homeassistant/components/youless/config_flow.py

```diff
@@ -0,0 +1,62 @@
+"""Config flow for youless integration."""
+import logging
+from urllib.error import HTTPError, URLError
+
+import voluptuous as vol
+from youless_api import YoulessAPI
+
+from homeassistant import config_entries, core, exceptions
+from homeassistant.const import CONF_DEVICE, CONF_HOST, CONF_NAME
+
+from .const import DOMAIN  # pylint:disable=unused-import
+
+_LOGGER = logging.getLogger(__name__)
+
+DATA_SCHEMA = vol.Schema({vol.Required(CONF_NAME): str, vol.Required(CONF_HOST): str})
+
+
+async def validate_input(hass: core.HomeAssistant, data):
+    """Validate the user input using the API connector."""
+    try:
+        api = YoulessAPI(data[CONF_HOST])
+        await hass.async_add_executor_job(api.initialize)
+    except HTTPError as exception:
+        raise CannotConnect() from exception
+    except URLError as exception:
+        raise CannotConnect() from exception
```

Comment:
Can be simplified:

```suggestion
    except (HTTPError, URLError) as exception:
        raise CannotConnect() from exception
```

Additionally:
 - Since this method is only called once, you might consider using the logic in place directly instead of a separate method.
 - There are no tests that covers the `HTTPError` case.

Reply:
I've moved the function into the `async_step_user` as advised. Good to know you can combine exceptions in one single catch in Python, I wasn't aware of that.

I will add additional tests for the case when an exception is thrown.